src/CPP/write_tree.cpp

```diff
@@ -101,4 +92,30 @@ std::string write_tree(const std::string &directory)
     compress_and_store(tree_hash.c_str(), tree_content, ".");
     // std::cout << "Tree hash for :" << directory << tree_hash << std::endl;
     return tree_hash;
-}
\ No newline at end of file
+}
+
+std::string check_file_perms(std::filesystem::perms file_perms)
+{
+    using std::filesystem::perms;
+    std::string ret = "100000 ";
+
+    if((perms::none != (file_perms & perms::owner_exec)))
```

Comment:
@akrisfx  Thanks a lot for your help.

### File Permissions in Git
Tracked Permissions:

Git only **tracks the executable bit** for files. This means it distinguishes between files that are executable and those that are not.
Git **does not track read/write permissions or ownership**. All files are considered to have read and write permissions for the user, group, and others (i.e., 0644 for non-executable files and 0755 for executable files).
Executable Bit:

If a file is marked as executable in the filesystem, Git stores this information.
When checking out a repository or switching branches, Git will apply the executable bit to files accordingly.

![image](https://github.com/ENAMINE1/MY-OWN-GIT/assets/111672103/6d9a42a4-9465-4d24-afdb-2cf6e38ddb28)

**The file _readme.md_  has only read permission for all the user groups but git sitll show 100644 permission.**

One thing that can be done is to **write a seperate function** that does file permission check, may be we can add this feature later. Also you can refactor the original code.

Keep up the excellent work. 🚀